runtimes/ocaml/runtime.ml

```diff
@@ -53,9 +53,34 @@ exception IndivisibleDurations
 exception ImpossibleDate
 exception NoValueProvided of source_position
 exception NotSameLength
+exception Division_by_zero (* Shadows the stdlib definition *)
 
-(* TODO: register exception printers for the above
-   (Printexc.register_printer) *)
+(* Register exceptions printers *)
```

Comment:
OK so you're going through OCaml's exception printing mechanism, which is more general. Why go through that very general mechanism that forces you to print source code positions in a crude way when you could catch in `interpreter.ml` the exceptions of `Runtime_ocaml` properly and access to their positions in a structured way (and that you can print beautifully)?

Reply:
At the moment, the position types in the runtime are different from the ones in the compiler, so we would need specific printing functions ; the interpreter doesn't seem like the right place to put those ?

Ideally, like for backtraces, we'd like unified printing for interpreter and runtime errors ; it would probably be best to start with unification of the underlying types. That's ongoing work but I wanted a quick fix: having a message with a usable position was the priority here.